### DISCUSSION TITLE
Color Picker as part of generator

### DISCUSSION THREAD

**Discussion by NYTEMODEONLY** [nftchef/art-engine#220] [Help + Q&A] [answered]
Hello All! First and foremost, thank you so much for this fork and continued development. I feel like this fork is exactly what I need for my project, but I have been having a troublesome time figuring out the last part - the config.js.

My project consists of robots and their colors are the very first "trait" - 7 total with some colors being more "rare" than others. Below is a picture of my file structure, ALL of the folders have the same content in them word-for-word. I am trying to get NFTCHEF's fork to select a color first, and continue generating as normal and circle back after completing one to start over again but I am at the point where I need to ask for help to achieve my end goal.

![ca2f0349d59f9f6c8b9473ea5e0c8bb3](https://user-images.githubusercontent.com/108635732/178375563-1a07f62f-263f-4e52-a33d-bb424ba0e2b5.png)

I have read through the github and dev website but cannot nail down the structure of the json language to achieve what I hope to achieve.

**tl;dr - Need layers in config.js to pick a color folder first, and generate an NFT as normal, circle back and do it again. 😅**

I've tried changing the below so much, but each time fails, or breaks or doesn't do what I'm trying to do and looking to see if anyone much more savvy than I at JSON can assist. **Please and thank you so much for your time!**

`const layerConfigurations = [
  {
    growEditionSizeTo: 10,
    namePrefix: "KRF", // Use to add a name to Metadata `name:`
    layersOrder: [
      { name: "COLORS" },
      { name: "CHASSIS" },
      {
        name: "HEAD",
        // options: {
        //   bypassDNA: true,
        // },
      },
      { name: "EXPRESSION" },
      { name: "ARMS" },
      { name: "ACCESSORIES" },
      { name: "UPGRADE" },
      { name: "CPU" },
    ],
  },`

**Comment by bezza123** [comment]
The code you have means that each of the traits should be in a top level folder not nested I think,

```
const layerConfigurations = [
  {
    growEditionSizeTo: 1000,
    namePrefix: "TradiesNFT", // Use to add a name to Metadata `name:`
    layersOrder: [
      { name: "Background"},
      { name: "Body" },
      { name: "Face"},
      { name: "Mouth" },
      { name: "Ears" },
      { name: "Hair" },
      { name: "Eyes" },
      { name: "Headwear" },
      { name: "Tool" },
    ],
  },
```
That is my config and all of those are top level folders, and they will add each layer in that order.

I have a nested structure in hair so that it only selects the correct facial hair to match main hair colour 
![image](https://user-images.githubusercontent.com/109074398/178524760-0127a37b-5ed7-434f-8c84-1f78a0513348.png)

But I dont list "Facial Hair" in the layers config but it still selects one and creates it as a "Facial Hair" trait 

```
"attributes": [
    {
      "trait_type": "Background",
      "value": "Light Green"
    },
    {
      "trait_type": "Body",
      "value": "Black Hoodie"
    },
    {
      "trait_type": "Face",
      "value": "Yellow Pencil"
    },
    {
      "trait_type": "Mouth",
      "value": "Oooo"
    },
    {
      "trait_type": "Ears",
      "value": "Ear 2"
    },
    {
      "trait_type": "Hair",
      "value": "Blond Ace"
    },
    {
      "trait_type": "Facial Hair",
      "value": "Blond Curly Mo"
    },
    {
      "trait_type": "Eyes",
      "value": "Eyes 5"
    },
    {
      "trait_type": "Headwear",
      "value": "Helmet Yellow"
    },
    {
      "trait_type": "Tool",
      "value": "Plaster"
    },
```

**Comment by nftchef** [comment]
if each folder of color maintains its own structure of parts (with required folders and weighted folders/files) you would want to have a single entry point folder in your config.js for all of those layers. (if you have any universal layers, they would be included outside of the `COLORS` folder and added to your layer config

e.g. the following setup 
![image](https://user-images.githubusercontent.com/91582112/178588065-66e027f4-e054-45dd-ad84-7e6acd763d4b.png)

```
    layersOrder: [
      { name: "COLORS"}
    ],

**Comment by NYTEMODEONLY** [comment, marked as answer]
For anyone who happens to land on this discussion looking for the same thing - I've found a temporary solution that I hope helps the next person.

In conjunction with nftchef's note and their guidance with using just the raw COLOR folder here:
`    layersOrder: [
      { name: "COLOR" },
    ],
  },`
  
**memonadil** on discord suggested simply making an additional folder called `BOT COLOR` inside each parent folder and put a single blank `.PNG` inside with it's respective name (e.g. BLACK#100.png).

This lets the generator just see, and scoop up that `trait` along the way giving each of my robots their respective colors as a trait during the process. Here is an example of my folder structure as well.

![image](https://user-images.githubusercontent.com/108635732/178884747-27603b46-5bec-46fd-bbde-a65d3716a683.png)

With proper z-indexing, all my folders are identical in names, layers get placed in the right order and all bots receive their appropriate color!

Thank You x1000!
